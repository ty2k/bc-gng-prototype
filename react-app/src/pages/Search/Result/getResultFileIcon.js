import Icon from "../../../components/Icon";

export default function getResultFileIcon(mimeType) {
  if (mimeType === "application/pdf") {
    return <Icon id={"file-pdf-solid.svg"} />;
  } else if (
    mimeType ===
    "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
  ) {
    return <Icon id={"file-word-solid.svg"} />;
  }
  return null;
}
