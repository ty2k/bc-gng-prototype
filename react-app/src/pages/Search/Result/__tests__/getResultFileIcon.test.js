import TestRenderer from "react-test-renderer";

import getResultFileIcon from "../getResultFileIcon";

import { mockResultPdf, mockResultWord } from "../__mocks__/mockResultData";

describe("getResultFileIcon", () => {
  const fileIconPdf = TestRenderer.create(
    getResultFileIcon(mockResultPdf?.$?.MIME, 1)
  ).toJSON();
  const fileIconWord = TestRenderer.create(
    getResultFileIcon(mockResultWord?.$?.MIME, 1)
  ).toJSON();

  it("renders a PDF icon correctly", () => {
    expect(fileIconPdf).toMatchSnapshot();
  });
  it("renders a Word icon correctly", () => {
    expect(fileIconWord).toMatchSnapshot();
  });
});
