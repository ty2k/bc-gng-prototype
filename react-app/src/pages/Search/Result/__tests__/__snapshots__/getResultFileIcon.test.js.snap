// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getResultFileIcon renders a PDF icon correctly 1`] = `
<svg>
  file-pdf-solid.svg
</svg>
`;

exports[`getResultFileIcon renders a Word icon correctly 1`] = `
<svg>
  file-word-solid.svg
</svg>
`;
